index.ts

```diff
@@ -9,97 +9,52 @@ export async function main() {
   // inputs
   const stainless_api_key = getInput('stainless_api_key', { required: true });
   const inputPath = getInput('input_path', { required: true });
+  const configPath = getInput('config_path', { required: false });
   const outputPath = getInput('output_path');
 
-  const decoratedSpec = await decorateSpec(inputPath, stainless_api_key);
-
-  if (outputPath) {
-    writeFile(outputPath, decoratedSpec);
-    info('Wrote spec to', outputPath);
-  }
-}
-
-type SignedUploadResponse = {
-  // Which key the upload is going to
-  fileKey: string;
-
-  // Which URL to POST an upload to
-  url: string;
-
-  // Which fields to include in the POST
-  fields: { [key: string]: string };
-};
-
-async function createSignedUpload(token: string): Promise<SignedUploadResponse> {
-  const response = await fetch('https://api.stainlessapi.com/api/spec/upload', {
-    method: 'POST',
-    headers: {
-      Authorization: `Bearer ${token}`,
-      'Content-Type': 'application/json',
-    },
-  });
-
+  info('Uploading spec and config files...');
+  const response = await uploadSpecAndConfig(inputPath, configPath, stainless_api_key);
   if (!response.ok) {
-    const errorMsg = `Failed to create a signed upload: ${response.statusText} ${response.text}`;
+    const errorMsg = `Failed to upload spec or config file: ${response.statusText} ${response.text}`;
     error(errorMsg);
     throw Error(errorMsg);
   }
 
-  return response.json() as unknown as SignedUploadResponse;
+  if (outputPath) {
+    const decoratedSpec = response.text();
```

Comment:
nit: I think you have to `await` this

Reply:
Surprised the type checker didn't report this issue.